### DISCUSSION TITLE
[Feature] Store public address for sharing images

### DISCUSSION THREAD

**Discussion by jjspierx** [immich-app/immich#5686] [Feature Request]
### The feature

I would like to be able to store an internal and external address in Immich somewhere, which could be used for sharing links to images.  

I access Immich exclusively through an internal ip address to my home network.  When I am away from home, I am connected to the internal network via VPN.  This works great for myself and for my immediate family. However, this makes it not possible to share links to images to my friends and other family.  I have a VPS setup with a reverse proxy that uses a VPN to tunnel into my home network.  I have yet to use this with Immich, but I would like to be able to use it for this purpose.  

I don't want to change how I connect to Immich internally, I just want to use it for links generated for sharing. This is because I am subject to bandwidth charges for traffic through my VPS.  If I could store an external/public facing address in Immich and have that used in the address generated for shared links I could have the best of both worlds.  Faster access, that isn't subject to bandwidth charges for myself and my immediate family, and easily shareable links to everybody else.

To implement this I would think one would just need to add a new property for the server called "PublicAddress" or something, and a then a bit property called something like "UsePublicAddressForShareLinks" or something like that.  Then when generating share links, if the UsePublicAddressForShareLinks = true, it would use that address when generating the link.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
You probably want to set up split dns, with that everything should just work. Closing as a duplicate of #4852

**Reply by jjspierx** [reply]
Split DNS would work and solve part of the problem, but it has a few caveats:

1. When my family members are not connected to the home LAN, all traffic to Immich will be directed through the VPS nginx reverse proxy, and be subject to bandwidth charges.

2. Setting up an internal DNS server with a split DNS configuration that every request made from my home network is using is pretty extreme when all that is needed is an easy way to store a public facing address that is used for link sharing.

I might be interested in giving this a go myself if you think others might find it useful.  I have never created a PR before on a project that wasn't my own, but I don't see this as being too difficult to implement.  If you don't think this is something that would be useful for the project, I understand.
